### PR TITLE
Return early, return often

### PR DESCRIPTION
**Pull request by justinmuskopf**
Addresses https://github.com/CFC-Servers/cfc_spawn_protection/issues/2

### CODE REVIEW

File: lua/autorun/server/sv_spawn_protection.lua [hunk 1]
```diff
@@ -109,14 +109,13 @@ end
...
     local spawnPoint = player.LinkedSpawnPoint
-    if ( spawnPoint and IsValid(spawnPoint) ) then
-        local spawnPointOwner = spawnPoint:CPPIGetOwner()
-        if ( spawnPointOwner != player ) then
-            return true
-        end
-    end
-	
-    return false
+    if not spawnPoint then return false end
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua]: Let's change this to an `or not` to combine this and the next line

File: lua/autorun/server/sv_spawn_protection.lua [hunk 2]
```diff
@@ -144,52 +143,43 @@ end
...
-
-            if ( !weaponIsAllowed( newWeapon ) ) then
-                removeSpawnProtection( player )
-                setPlayerVisible( player )
-                removeDecayTimer( player )
-                removeDelayedRemoveTimer( player )
-            end
-        end
-    end
+    removeSpawnProtection( player )
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua]: Is this indentation correct? Looks 4 off to me

File: lua/autorun/server/sv_spawn_protection.lua [hunk 3]
```diff
@@ -144,52 +143,43 @@ end
...
+    setPlayerVisible( player )
+    removeDecayTimer( player )
+    removeDelayedRemoveTimer( player )
 end
 
 -- Called on player keyDown events to check if a movement key was pressed
 -- and remove spawn protection if so
 local function spawnProtectionMoveCheck( player, keyCode )
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua]: More indentation weirdness in this method

File: lua/autorun/server/sv_spawn_protection.lua [hunk 4]
```diff
@@ -144,52 +143,43 @@ end
...
 
-        if ( playerHasSpawnProtection( player ) ) then
-            local playerIsMovingThemselves = keyVoidsSpawnProtection( keyCode )
-
-            if ( playerIsMovingThemselves ) then
-                delayRemoveSpawnProtection( player )
-            end
-        end
+    local playerIsMovingThemselves = keyVoidsSpawnProtection( keyCode )
+	if ( playerIsMovingThemselves ) then
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua]: `if playerIsMovingThemselves then delayRemoveSpawnProtection( player ) end`

File: lua/autorun/server/sv_spawn_protection.lua (hunk 3 above)
Comment (justinmuskopf) [reply lua/autorun/server/sv_spawn_protection.lua]: Man, I saw that too. 
The vim boy is set to four spaces, though. 
I'll check 'er out

Comment (brandonsturgeon) [reply lua/autorun/server/sv_spawn_protection.lua]: `gg=G`

**Comment by brandonsturgeon** [comment]
We should test this branch before merging. Tonight?

File: lua/autorun/server/sv_spawn_protection.lua [hunk 5]
```diff
@@ -99,7 +99,7 @@ local function createDelayedRemoveTimer( player )
         removeDecayTimer( player )
     end)
 end
-https://gist.github.com/Dinnerbone/943fbcd763c19be188ed6b72a12d7e65
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_spawn_protection.lua]: 🤔

**Pull request merged**